Sweetener.Logging/Loggers/Logger.cs

```diff
@@ -0,0 +1,162 @@
+        protected Logger()
+            : this(LogLevel.Debug, null)
+        { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Logger"/> class with a minimum
+        /// logging level for the current culture.
+        /// </summary>
+        /// <param name="minimumLevel">The minimum level of log requests that will be fulfilled.</param>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="minimumLevel"/> is unrecognized.</exception>
+        protected Logger(LogLevel minimumLevel)
+            : this(minimumLevel, null)
+        { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Logger"/> class with a minimum
+        /// logging level and an <see cref="IFormatProvider"/> object for a specific culture.
+        /// </summary>
+        /// <remarks>
+        /// If <paramref name="formatProvider"/> is <c>null</c>, the formatting of the current culture is used.
+        /// </remarks>
+        /// <param name="minimumLevel">The minimum level of log requests that will be fulfilled.</param>
+        /// <param name="formatProvider">An <see cref="IFormatProvider"/> object for a specific culture.</param>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="minimumLevel"/> is unrecognized.</exception>
+        protected Logger(LogLevel minimumLevel, IFormatProvider formatProvider)
+            : this(minimumLevel, formatProvider, LogPattern.Default)
+        { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Logger"/> class with a minimum
+        /// logging level and an <see cref="IFormatProvider"/> object for a specific culture.
+        /// </summary>
+        /// <remarks>
+        /// If <paramref name="formatProvider"/> is <c>null</c>, the formatting of the current culture is used.
+        /// </remarks>
+        /// <param name="minimumLevel">The minimum level of log requests that will be fulfilled.</param>
+        /// <param name="formatProvider">An <see cref="IFormatProvider"/> object for a specific culture.</param>
+        /// <param name="pattern">A format string that describes the layout of each log entry.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="pattern"/> is <c>null</c>.</exception>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="minimumLevel"/> is unrecognized.</exception>
+        /// <exception cref="FormatException">The <paramref name="pattern"/> is not formatted correctly.</exception>
+        protected Logger(LogLevel minimumLevel, IFormatProvider formatProvider, string pattern)
+        {
+            if (minimumLevel < LogLevel.Debug || minimumLevel > LogLevel.Fatal)
+                throw new ArgumentOutOfRangeException(nameof(minimumLevel), $"Unknown {nameof(LogLevel)} value '{minimumLevel}'");
+
+            if (formatProvider == null)
+                formatProvider = CultureInfo.CurrentCulture;
```

Comment:
There is precedence for this behavior in [`TextWriter.cs`](https://github.com/dotnet/corefx/blob/6f9570e012054bf407778c0b07632e7733ecbb13/src/Common/src/CoreLib/System/IO/TextWriter.cs#L57)